swarm/pss/pss.go

```diff
@@ -131,7 +134,10 @@ func (self *Pss) String() string {
 //
 // In addition to params, it takes a swarm network overlay
 // and a DPA storage for message cache storage.
-func NewPss(k network.Overlay, params *PssParams) *Pss {
+func NewPss(k network.Overlay, params *PssParams) (*Pss, error) {
+	if params.privateKey == nil {
```

Comment:
what happens if i want to instantiate pss without a private key since i want to just use `raw` messages and handle the encryption on my side?

Reply:
You'd still set the key. Other protocols might want to use encryption.